### DISCUSSION TITLE
How can one search hosts by tags?

### DISCUSSION THREAD

**Discussion by budachst** [CrowdStrike/falconpy#989] [Q&A] [answered]
Is there a possibility to search hosts by tags? I mainly see the "plain" host attributes as filter parameters in the API documentation, but not tags.

**Comment by jshcodes** [comment, marked as answer]
You should be able to do a wildcard search as follows:

```python
from falconpy import hosts

hosts = Hosts(client_id=CLIENT_ID, client_secret=CLIENT_SECRET)

search_string = "whatever"

result = hosts.query_devices_by_filter_scroll(filter=f"tags:*'*{search_string}*'")
```

You can also adjust this to perform an exact match:

```python
result = hosts.query_devices_by_filter_scroll(filter=f"tags:['{search_string}']")
```

**Reply by jshcodes** [reply]
Follow up: I am creating an issue to add this missing filter option to the [falconpy.io](https://falconpy.io) documentation. 😄

**Reply by budachst** [reply]
Great - thanks @jshcodes - this will come very handy…

**Reply by budachst** [reply]
I am wondering, if this can also be used to query for the absence of a tag. If not, I will have to run a loop over the selected hosts to check that seperately. My guess is, that a check for the absence if a tag would only work on hosts, which only would carry this tag and no other ever, right?

**Reply by jshcodes** [reply]
Try this filter:

```python
result = hosts.query_devices_by_filter_scroll(filter="tags:null")
```
